### ISSUE TITLE
Changes required for using the Helm-direct plugin 

### ISSUE THREAD

**Issue by antgamdia** [kubeapps/kubeapps#3165]
As #3085 is increasingly growing, for the sake of readability, I'm splitting it into a separate issue, so that we can keep track of the required changes.

### Quick summary (sep 1st):

Triggering the release

### Quick summary (aug 31th):

Bugs detected after CI, working on that.

- [x] Multicluster mode failing, unknown issues, check later.
- [x] Upgrading a chart. Twofold scenario:  chart w/ repo; chart w/o repo.
- [ ] Getting duplicates charts after searching for chart names in input

### Quick summary (aug 30th):

Getting CI passing, not luck

### Quick summary (aug 27th):

- Kubeapps v2.3.4 released!
- Issues with the CI
- Minor PRs
- Fixing tests: first round completed! 

However, 20 tests from 5 test suites are still being skipped, (below). The reasons range from: 1) tradeoff complexity vs velocity; 2) something requiring extra checks because could be actual errors; 3) candidate to be removed, but unsure.

If you can continue from here, @absoludity , it'd be great. Otherwise, I'll continue working next week

1. - src/components/Catalog/Catalog.test.tsx (9)
2. - src/UpgradeForm.test.tsx (6)
3. - src/actions/apps.test.tsx (3)
4. - src/shared/App.test.ts (1)
5. - src/components/AppList/AppListItem.test.tsx (1)

Edit: I'm also running the e2e tests to see what is failing (**important**: we should also check the Bitnami tests, otherwise, a failing test there will block the release). I'll re-run them to ensure they're not false positives.

```console
 
FAIL  use-cases/07-upgrade.js (7.508 s)

 RUNS  ...
  ● Upgrades an application

    Option not found "select[name="chart-versions"]" ("7.3.2")


FAIL  use-cases/08-rollback.js (67.947 s)

 RUNS  ...
  ● Upgrades an application

    TimeoutError: Text not found "(current: 3)"
```

### Quick summary (aug 26th):

- Fixing issues when releasing v2.3.4
- Fixing tests

There are 2 tests suites (56 tests) which are failing (via `yarn react-scripts test -- --watchAll=false 2>&1 | grep 'FAIL' | sort --unique`) + 7 tests commented out so far:

1. FAIL src/components/Catalog/Catalog.test.tsx (30 tests)
2. FAIL src/components/ChartView/ChartView.test.tsx (10 tests)
4. FAIL src/components/UpgradeForm/UpgradeForm.test.tsx (16 tests)

### Quick summary (aug 25th):

- Triggering the v2.3.4 release
- For the sake of simplicity and for you all to know, I'm pushing my changes to the latest branch (not others) https://github.com/kubeapps/kubeapps/pull/3271.

There are 9 tests suites (83 tests) which are failing (via `yarn react-scripts test -- --watchAll=false 2>&1 | grep 'FAIL' | sort --unique`) + 7 tests commented out so far:

1. FAIL src/actions/charts.test.tsx
2. FAIL src/components/AppView/ChartInfo/ChartInfo.test.tsx (6.543 s)
3. FAIL src/components/Catalog/Catalog.test.tsx (10.67 s)
4. FAIL src/components/ChartView/ChartHeader.test.tsx
5. FAIL src/components/ChartView/ChartReadme.test.tsx (5.213 s)
6. FAIL src/components/ChartView/ChartView.test.tsx (8.615 s)
7. FAIL src/components/DeploymentForm/DeploymentForm.test.tsx (6.493 s)
8. FAIL src/components/DeploymentFormBody/DeploymentFormBody.test.tsx (5.882 s)
9. FAIL src/components/UpgradeForm/UpgradeForm.test.tsx (6.735 s)

### Quick summary (aug 24th):

I'd say I've mostly finished every required change to make Kubeapps usable again :tada: 
The changes are ready to be pulled at: https://github.com/kubeapps/kubeapps/tree/ui-newapi-2-installedPackages

- [x] Add the new `version` field everywhere. Now both the "installed apps" view and the "installed app details" view are able to display the version upgrade message properly
- [x] add missing fields (home, URL, etc)
- [x] fix a trivial bug in getAvailableVersions 
- [x] add `cluster` parameter everywhere 
- [x] Allow rollbacking apps using the revision number as part of the id
- [x]  Add shortDescription and icon via query to InstalledPacakgeDetail.AvailablePackageRef.ID
- [x] Removed Helm icon from the right panel
- [x] Use the existing behavior to show the "deployed resources" in the installed view

To sum up, here's a list of pending things (as of today)

1. [priority/higest] Get the tests to work again. I'd suggest a twofold approach:
    1.  Ensure every e2e is passing ASAP
    2.  Start fixing the failing unit tests, if too complex, skip. Prioritize coverage over test quality
2. [priority/high] Implement operations for the Helm actions, namely:
    1. CreateInstalledPackage
    2. UpdateInstalledPackage
    3. DeleteInstalledPackage 
3. [priority/medium] Implement the corresponding core operations for each op already implemented in the Helm direct plugin (I'd suggest something quick and then think about the pagination and dups handling later)
4. [priority/low] Rename all the references to "charts" in the code to "package" or sth similar. Do it in a separate PR for reducing noise.

Comparison: new (left) vs old (right) versions. 

https://user-images.githubusercontent.com/11535726/130686143-123f9e3c-ddc3-45cd-9343-2f9cc5ebd2fd.mp4

Resources are not being updated because of the wrong proxied websocket connection through the react dev tools.

### Quick summary (aug 23th):

After my week off: today I've merged the changes (in master) into my open branches, as well as I've merged those trivial PRs. I'm starting to work on adding the missing fields you've been working on this week (thanks!). 

### Quick summary (aug 12th):

Everything splitted into nine PRs:

- [x] https://github.com/kubeapps/kubeapps/pull/3262
- [x] https://github.com/kubeapps/kubeapps/pull/3264
- [x] https://github.com/kubeapps/kubeapps/pull/3265
- [ ] https://github.com/kubeapps/kubeapps/pull/3266
- [ ] https://github.com/kubeapps/kubeapps/pull/3267
- [ ] https://github.com/kubeapps/kubeapps/pull/3268
- [ ] https://github.com/kubeapps/kubeapps/pull/3269
- [ ] https://github.com/kubeapps/kubeapps/pull/3270
- [ ] https://github.com/kubeapps/kubeapps/pull/3271

Some notes about them:

- PRs 1-3 are pretty straightforward and can be easily merged  (3 into 2 into 1 into our dev branch)
- PRs 4.x are when the tests start failing, it's normal since removing the containers implies some changes in the way the tests are performed. I've created those PRs just to be able to work in each (failing) test suite separately. Even, hopefully, in parallel :pray: 
    -  There is no actual need to get everything perfectly working here, since  PRs 5-6 will override some changes.
- PR 5 is the first bulk change towards using available packages API. It has implications in other views, so it cannot be splitted more IMHO. The base branch I chose is a 'fake' one supposing 4.1..4.4 have been merged. This is just intended for avoiding a much bigger review.
- PR 6 Created over PR5, it includes the code to use the installed packages API. It also removes old references to the `assetsvc` code, so this component can now be safely disconnected (indeed, I tried it and it works).

### Quick summary (aug 11th):

What's not ready yet?

 1. In _installed packages summary_ view: 
      - retrieve updates on app versions
 2. In _installed package details_ view:
      - retrieve updates on versions
      - print icon
      - fetch generated resources

And, of course, testing and the PRs.

Video:

https://user-images.githubusercontent.com/11535726/129124495-23e363ca-0d28-4cc3-97f9-4972fbb2d696.mp4

![Selection_001](https://user-images.githubusercontent.com/11535726/129120626-b6214021-c739-400d-8686-a3dcd3e5cf96.png)

If curious, current changes in a local PR: https://github.com/antgamdia/kubeapps/pull/393/files

### Quick summary (aug 10th):

All the changes related to available packages (summaries and details) have been implemented. This ranges from querying the chart details to prepare the values+schema to be deployed, and also retrieving the info when upgrading.

What's not ready yet?

 1. installed packages summary view  (I will work soon in this view, trying to perform the changes in a separate PR)
 2. installed package details view (depends on https://github.com/kubeapps/kubeapps/pull/3251 to be merged)

Then, the actual deploy/rollback/delete/etc endpoints should be updated once they're available, but not yet.

Upcoming plans:

1. Continue working on the installed packages views (summary+details). Hopefully, in a separate PR if there are no dependencies with the rest of my changes
2. Start creating the draft PRs (if curious, here is a local PR for you to have a look at the changes https://github.com/antgamdia/kubeapps/pull/391/files). I really like to split it into different PRs, at least a) removing containers+pruning old code; b) actual changes.
3. Start working on fixing the (huge amount) of failing tests
4. Make the PR ready for review (if I can't get the tests done in time, at least you'll have the draft)
5. Discuss the required changes in the backend. They can be addressed in follow-up PRs as almost everything has been worked around.

Yet to be decided: https://github.com/kubeapps/kubeapps/issues/3165#issuecomment-882943900

Overview at: https://github.com/kubeapps/kubeapps/issues/3165#issuecomment-882950182

Demo:

https://user-images.githubusercontent.com/11535726/128924907-d269576a-e689-4cd1-8feb-c26a56f1b614.mp4

https://user-images.githubusercontent.com/11535726/130686073-3a875b48-0a57-4833-a965-a9a2780fdc83.mp4

**Comment by antgamdia** [comment]
## Missing fields

- [ ] Add `latestAppVersion` to `GetAvailablePackageSummaries`.
    - **Priority**: high
    - **Status**: to be considered as a required filed in the summary and detail responses
    - **Core API relationship**: this field should be available for every plugin
    - **Reason**:  the current cards in the catalog show the appVersion (not the chart version)

- [ ] Add `category` to `GetAvailablePackageSummaries`.
    - **Priority**: high
    - **Status**: to be considered as an optional field in the summary and detail responses
    - **Core API relationship**:  this field is not available for every plugin
    - **Reason**: it is required for filtering by category in the UI. 
   
- [ ] Add `repo` to `GetAvailablePackageSummaries` and `GetAvailablePackageDetails`.
    - **Priority**: low (can be workarounded)
    - **Status**: to be considered as an optional field in the summary and detail responses
    - **Core API relationship**: this field is not available for every plugin
    - **Reason**: it is required for filtering by repo in the UI. This functionality is used in both the catalog view (as a filter) and the AppRepo view (as a link to the catalog).

**Comment by antgamdia** [comment]
## Missing operations

TBD

**Comment by antgamdia** [comment]
## UI (non-visible) changes

### Change-as-you-go
- [ ] Rename "charts" references with "availablePackage"/"installedPackage" as we are modifying the code. Split it into two PRs for separating trivial changes from important ones.
- [ ] Update the tests accordingly as we modify the code.

### Additional required changes
- [ ] Create a "base test case" for the grpc client
- [ ] Improve the error handling in the grpc client (we had `AxiosWithAuth`, but now this logic should be addressed somehow)

**Comment by antgamdia** [comment]
_Intentionally blank post for future usage_

**Comment by antgamdia** [comment]
## List of use cases

### Catalog overview
- [ ] List packages
- [ ] Filter packages by text (seach query)
- [ ] Filter packages by checkbox:
  - [ ] Type
  - [ ] Repository
  - [ ] Category
  - [ ] Operator provider

### Package details
- [ ] TBD

### Package actions
- [ ] Deploy

### Installed package details
- [ ] TBD

### Installed package actions
- [ ] Upgrade
- [ ] Rollback
- [ ] Delete

### Repositories
_(out of scope)_

**Comment by absoludity** [comment]
Why not have all the checkboxes in the main issue description?

**Comment by antgamdia** [comment]
> Why not have all the checkboxes in the main issue description?

I did consider that option, but wanted to have an independent link for each point (for instance, when writing a PR description). Otherwise, it's easy to get lost in a really long post (because AFAIK is not possible to link to a title header inside an issue comment, isn't it?)

**Comment by antgamdia** [comment]
Question here, @absoludity : I'm facing some self-inconsistencies  while naming vars, so I'd rather pose the question here to make it explicit for future references.

Is it:
1. (plural) _AvailablePackageSummaries_ or _AvailablePackage**s**Summaries_ 
2. (singular) _AvailablePackageSummary_ or _AvailablePackage**s**Summary_ 

?

**Comment by absoludity** [comment]
> Question here, @absoludity : I'm facing some self-inconsistencies while naming vars, so I'd rather pose the question here to make it explicit for future reference.
> 
> Is it:
> 
>     1. (plural casel) _AvailablePackageSummaries_ or _AvailablePackage**s**Summaries_

The first one - _AvailablePackageSummaries_ . Sorry if I've inadvertently added an extra s somewhere (though I can't see it in the doc or proto with the extra s?)

> 
>     2. (singular case) _AvailablePackageSummary_ or _AvailablePackage**s**Summary_

Definitely _AvailablePackageSummary_ here for the singular.

> 
> ?
> 
> Edit: one more question:
> 
> Adding `available package identifier name` to the `AvailablePackageReference` spec: right now we store the `identifier` (see below). There is no particular reason, as the UI can work as is, but I wonder if adding directly the `name` here would simplify the UI logic later on.

You mean adding the name to the `AvailablePackageReference`, even though it's already on the message (such as the `AvailablePackageSummary` or `Detail`)? How or why do you think this would simplify the UI?

The reference should just be the info that uniquely identifies the package. We did use the name originally but realised it was ambiguous for some packaging formats such as Helm (this was discussed in the doc), and so used an Identifier instead, and added the `name` to the message itself.

> Currently, we are splitting by the `/` character and getting the first part (as in `bitnami/kubeapps --> bitnami`)

I'm not sure what you mean here. The UI should be using  the name from the actual summary message. Actually, I see we added it to the proto, but didn't update the AvailablePackageSummaryFromChart to include it. Let me update that.

**Comment by antgamdia** [comment]
> The first one - _AvailablePackageSummaries_ . Sorry if I've inadvertently added an extra s somewhere (though I can't see it in 
> Definitely _AvailablePackageSummary_ here for the singular.

Great. No, it was my bad probably during my former PR. I'll squeeze this change in my open PR to make sure we use  _AvailablePackageSummary_ and _AvailablePackageSummaries_. Thanks for clarifying!

> You mean adding the name to the `AvailablePackageReference`, even though it's already on the message (such as the `AvailablePackageSummary` or `Detail`)? How or why do you think this would simplify the UI?

Well, maybe it's just me I'm not getting it. From the UI point of view, what I need is just the _repository name_, so that we can continue filtering as we do now. 

Currently, we are [using the chart ID](https://github.com/kubeapps/kubeapps/blob/master/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go#L263) (`foo/bar`), so, in the UI code, I'm doing `c.availablePackageRef.identifier.split("/")[0]` for getting the repository name.

Not an issue, of course, it's doable and, in fact, it works, My comment was just wondering if we ever would want this `repository name` concept in our current model. But, perhaps let's move this discussion to the design doc, as it is not any urgent field/operation that we really need now.

> I'm not sure what you mean here. The UI should be using the name from the actual summary message. Actually, I see we added it to the proto, but didn't update the AvailablePackageSummaryFromChart to include it. Let me update that: #3172

Yes, I don't have any issue with the _package name_ but with the _repository name._ I didn't explain it well, let me extract the current mapping I've done:

```js
{
  type: "chart",
  id: `chart/${c.availablePackageRef?.identifier}`,
  item: {
    id: `chart/${c.availablePackageRef?.identifier}/${c.latestPkgVersion}`,
    name: c.displayName,
    icon: c.iconUrl,
    version: c.latestAppVersion,
    description: c.shortDescription,
    repo: { name: c.availablePackageRef?.identifier.split("/")[0] } as IRepo, // <<- here is where I need the repo name
}

```

**Comment by absoludity** [comment]
> Yes, I don't have any issue with the _package name_ but with the _repository name._ I didn't explain it well,

Ah - ok. Yes, I wasn't sure which you meant.

> let me extract the current mapping I've done:
> 
> ```js
> {
>   type: "chart",
>   id: `chart/${c.availablePackageRef?.identifier}`,
>   item: {
>     id: `chart/${c.availablePackageRef?.identifier}/${c.latestPkgVersion}`,
>     name: c.displayName,
>     icon: c.iconUrl,
>     version: c.latestAppVersion,
>     description: c.shortDescription,
>     repo: { name: c.availablePackageRef?.identifier.split("/")[0] } as IRepo, // <<- here is where I need the repo name

I see, and this is because the UI displays the repo name with each tile. Makes sense. Yes, let's go with that at the moment, but I agree - the UX shouldn't (and can't generally) need to know how to extract the name of the repository from the package identifier. +1 to add a repository display name to the summary data (not the ref) for now. And the detail, though we may later include fuller info about the repo on the detail once we've defined repos in the API etc.

> }
> ```

**Comment by antgamdia** [comment]
> * [ ]  Update the tests accordingly as we modify the code.
> * [ ]  Create a "base test case" for the grpc client

Well, some fresh updates here:

Today I've been trying to get the tests to work. Until now, the approach was mocking the lowest level component when making an HTTP request. It was really easy as we were using Axios, which is a proven solution with many adopters.

With that in mind, I tried to replicate this approach with our new GrpcClient. It was promising since I noticed there is a fake transport for testing (https://www.npmjs.com/package/@improbable-eng/grpc-web-fake-transport)

But, when trying to actually fake some messages... I was not that good... 
To sum up:
`grpc-web-fake-transport` requires messaged which type is defined by `google-protobuf` (we aren't using this lib). As this type is defined as an abstract with plenty of things to satisfy... I decided to give a try to (yet) another generator (`ts-protoc-gen`), but it is mixing js and ts and it's a mess for performing the imports. 
Next, I saw there is a fork (`protoc-gen-ts`) that claims to be typescript-ready. Well, it wasn't working in our Jest environment (if a file is only made of types, you have to add the `declare` keyword in Typescript, I just discovered it today). So... I manually edited the generated file (not a good idea).
After these tweaks, I was able to build the message and set some fields (hooray!) [that is, the tests seem to be working!].
But... I don't why, but when adding arrays... I run into a weird error (the only reference I found is from a [SO crawler,](http://5.9.10.113/63595871/google-protocol-buffers-typeerror-ce-toarray-is-not-a-function-when-setting) since the post was deleted).

So, at this point I just gave up... so instead of mocking the HTTP client, I just mocked the method which performs the HTTP and returns the data (main changes in commit https://github.com/kubeapps/kubeapps/pull/3164/commits/8d1230fa96e41d70df1f2ef63180232af77d7a68)

**Comment by absoludity** [comment]
> Until now, the approach was mocking the lowest level component when making an HTTP request. It was really easy as we were using Axios, which is a proven solution with many adopters.

This was (I assume) because we weren't using a client, but were using Axios directly from our code, therefore to test that we need to mock at that level (the URL foo is called and the response is bar.

> So, at this point I just gave up... so instead of mocking the HTTP client, I just mocked the method which performs the HTTP and returns the data (main changes in commit [8d1230f](https://github.com/kubeapps/kubeapps/commit/8d1230fa96e41d70df1f2ef63180232af77d7a68)) and, well, good news finally, it worked. I just don't know if this approach is OK or we should invest more effort in the former.

Right, given that we now have a client provided that we can trust, I think it's totally appropriate that we trust the client and have our tests assume results directly from the client, as you've done here (I think?). This is better on a number of levels:

- We're not testing external code from the client itself,
- If we update implementation details in the grpc calls, such as the URLs or http endpoints (grpc names), the UX tests aren't brittle and won't break, as they depend only on the behavior of the client.

Our unit tests should be testing our code only, which I think is the direction you've headed here? Great.

**Comment by antgamdia** [comment]
> Our unit tests should be testing our code only, which I think is the direction you've headed here? Great.

Great, really happy to hear that. I spent too much time trying to come up with a solution to overcome the problem and it was really frustrating haha. On the other hand, it is also a (kind of) risk we assumed when moving to the grpcweb approach. This stack is not so popular as directly calling rest apis is. 

Hopefully, implementing subsequent operations would be easier once we got that sorted out :)

**Comment by antgamdia** [comment]
Thanks for your comments and for adding the missing fields :) Continuing here the discussion about each point:

-  Add concept of "deployed resources" to `InstalledPacakgeDetails`
    > Can we use the existing endpoint for these in the meantime, while we plan to add a separate endpoint (not part of `InstalledPackageDetails`).

Yes, I think it's the best option right now. I have to figure out how to do that (I'm still a bit out of context), but I'll give it a try.

-  Add concept of "revision" to `InstalledPacakgeDetails`
> This is quite Helm specific (assuming you mean the release revision), and more generally, specific to each packaging system. So more generally, it'll be an ordered list of string rollback targets, I think? If these are included with the InstalledPackageDetail as packaging-system dependent identifiers for rollback targets ( `repeated string` which is `[]string`, for helm, it'll just be the revisions `["3", "2", "1"]` or similar) to present to the user for selection and passing to the rollback api call. 

Good idea indeed! So it would be required for the release, I guess. Let's discuss this point with @gfichtenholt and see what he thinks.

- Add `shortDescription` to `InstalledPacakgeDetail`
> This is part of the `AvailablePackageDetail` which should be queried from the `InstalledPackageDetail`? (That is, we are viewing a single package here, is there a reason you think you shouldn't be grabbing the available package detail for this info, using the `InstalledPackageDetail.AvailablePackageRef` ?)

Yes, it is a possibility I have thought of... but it would imply an additional API call (as the `AvailablePackageRef` is obviously just the ref) generating something we should keep in the redux state. It's doable and indeed makes sense. I just performed the immediate and direct changes. I'll give it a shot. 
Indeed, I've noticed I also need the icon URL here as well.

- Handle duplicates in `AvailablePackageSummaries`.
> I'm unable to reproduce this? See an [example gist](https://gist.github.com/absoludity/bcb3fcffd3b1e86d28cf06d590889caf) showing that the returned available packages are only for the namespace in which they are defined. If I add a repository to the `default` namespace, I would expect to see duplicates - perhaps that's the situation you mean? Let me know what's missing.

Exactly, that's was my example. I was wrongly filtering by package id (eg bitnami/kubeapps) and was getting inconsistent results. But you're right... dups are expected in this situation... so I'll remove the filter.

- Query also over global namespace (or at least, optionally do that) in `AvailablePackageDetail` or add search by name in summaries,
> I'm confused about this one: I assume you don't mean when displaying the `InstalledPackageDetail` because in that case you already have the available package reference. So it must be when checking chart updates for `InstalledPackageSummary`'s? But in that case, you already have the latest pkg version with each summary. Let's chat about this one. 

Now I'm also confused, don't remember what was the actual issue. I will ping you when I reproduce what I meant in my original message. However, I do have a question: what should be the logic for displaying a new version to the users?
I mean, currently, we display a message whether the chart (aka pkg) version or the app version is updated. Now we have the "matching version" concept. Some ideas here:

1) Prioritize "matching versions" over "all versions", that is: user is in `1.0.0` and there are `1.1.0` and `2.0.0`, the user will be prompted twice. "Update 1.0.0->1.1.0, and, later on, update 1.1.0->2.0.0
2) Display both "matching" and "all" version updates (it would require some UI changes).
3) Ignore this "matching" field, and continue using the current approach.

I'd go with 3) for the moment to keep the current behavior, but see what you think.

**Comment by absoludity** [comment]
> However, I do have a question: what should be the logic for displaying a new version to the users?
> I mean, currently, we display a message whether the chart (aka pkg) version or the app version is updated. Now we have the "matching version" concept.

The matching version is not relevant for Helm since with helm, a specific version (rather than a range of versions) is specified for install, so we can ignore it right now, but yes, for the future (when we're displaying flux packages)...

> Some ideas here:
> 
>     1. Prioritize "matching versions" over "all versions", that is: user is in `1.0.0` and there are `1.1.0` and `2.0.0`, the user will be prompted twice. "Update 1.0.0->1.1.0, and, later on, update 1.1.0->2.0.0
> 
>     2. Display both "matching" and "all" version updates (it would require some UI changes).

Initially (in the future), I would display matching versions only for packages that define a range of versions. They will normally be handled by the packaging system itself anyway (unless the reconciliation is paused for that package). We could later update to also provide information about other versions, but the user would (and should) need to update the installed package version range to get that update - it shouldn't be just clicking to upgrade IMO, since it is modifying the declared intent. Let's think about that more at the time.

> 
>     3. Ignore this "matching" field, and continue using the current approach.
> 
> 
> I'd go with 3) for the moment to keep the current behavior, but see what you think.

Yes, 3 for now. Longer term, we need

**Comment by antgamdia** [comment]
> Yes, 3 for now. Longer term, we need

Great. Another follow-up question: what should we show to the users? Supposing there are both a package and an app upgrade, should we display both? Or should we maintain the existing order "appVersion over pkgVersion" ?

1- Idea: 
![image](https://user-images.githubusercontent.com/11535726/130630642-9754c9b3-ef7d-4034-ab02-1b872c7709fa.png)

2- Existing (well, the _now using x.xx_ is mine)
![image](https://user-images.githubusercontent.com/11535726/130631419-cc3c1b3b-24fb-45f4-8577-15e79d95e564.png)

3- Current one:
![image](https://user-images.githubusercontent.com/11535726/130631886-4b69b319-2b10-4e58-ae9d-fef4947c7463.png)


For preserving the current behavior, I'll go with 3 unless you prefer otherwise :)

**Comment by ppbaena** [comment]
+1 to the 3rd option. 

IMO its clear from UX perspective. App version and package version together in the same tooltip could be confusing (specially as it is something that always will happen for new app version).

**Comment by antgamdia** [comment]
Closing in favor of https://github.com/kubeapps/kubeapps/issues/3340
